src/test/java/bio/terra/service/dataset/DatasetIntegrationTest.java

```diff
@@ -456,6 +460,56 @@ public void testSoftDeleteNotInFullView() throws Exception {
     assertSnapshotTableCount(bigQueryLess, snapshotLess, "sample", 5L);
   }
 
+  @Test
+  public void testCombinedMetadataDataIngest() throws Exception {
+    DatasetSummaryModel datasetSummaryModel =
+        dataRepoFixtures.createDataset(steward(), profileId, "dataset-ingest-combined.json");
+    UUID datasetId = datasetSummaryModel.getId();
+
+    IngestRequestModel ingestRequest =
+        new IngestRequestModel()
+            .format(IngestRequestModel.FormatEnum.JSON)
+            .ignoreUnknownValues(false)
+            .maxBadRecords(0)
+            .table("sample_vcf")
+            .path("gs://jade-testdata-useastregion/dataset-ingest-combined-control.json");
+
+    IngestResponseModel ingestResponse =
+        dataRepoFixtures.ingestJsonData(steward(), datasetId, ingestRequest);
+
+    var loadResult = ingestResponse.getLoadResult();
+
+    assertThat("file_load_results are in the response", loadResult, notNullValue());
+
+    assertThat(
+        "all 4 files got ingested", loadResult.getLoadSummary().getSucceededFiles(), equalTo(4));
+
+    var fileModels =
+        loadResult.getLoadFileResults().stream()
+            .map(BulkLoadFileResultModel::getTargetPath)
+            .map(
+                path -> {
+                  try {
+                    return dataRepoFixtures.getFileByName(steward(), datasetId, path);
+                  } catch (Exception e) {
+                    return null;
+                  }
+                })
+            .flatMap(file -> Optional.ofNullable(file).stream())
+            .collect(Collectors.toList());
+
+    var fileIds =
+        loadResult.getLoadFileResults().stream()
+            .map(BulkLoadFileResultModel::getFileId)
+            .flatMap(id -> Optional.ofNullable(id).stream())
```

Comment:
If you're trying to filter null values from a stream, a better way to do it is
```suggestion
            .filter(Objects::nonNull)
```